### DISCUSSION TITLE
Can Low-code connector  read a simple JSON Array that is not contained in an object

### DISCUSSION THREAD

**Discussion by mike-azility** [airbytehq/airbyte#33995] [Connector Questions]
We need to extract data from a very simple API it just returns the JSON response  `["string1", "string2", "string3"  .... ] ` etc   .  When we try to access it using the low code connector we get a `TypeError: string indices must be integer` error. 

 I created a very simple Python server to test different responses,  it seems the connector would only work if the array contained Objects not strings.    

Is this expected and what other methods could we use to access this API?  
Thanks in advance
Mike

**Comment by natikgadzhi** [comment]
Hey @mike-azility! 

Here's a related issue we've looked at with the team: https://github.com/airbytehq/airbyte/issues/33017 

You're right that currently Airbyte expects the API response to be a JSON array of objects, not array of strings. It's possible to add a transformation option to Airbyte CDK that would allow string values instead of objects, we just haven't gotten to this yet.

**Comment by mark-fzks** [comment]
Hi, I am also facing the same issue. This is necessary, as the CrowdStrike API only returns an array of strings as resources.
https://www.crowdstrike.com/
```
[
  {
    "meta": {
      "query_time": 0.00xxxxxx,
      "pagination": {
        "offset": 0,
        "limit": xxx,
        "total": xx
      },
      "powered_by": "cs.exampleapi",
      "trace_id": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"
    },
    "resources": [
      "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx",
      "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"
    ]
  }
]
```

**Comment by brianbolt** [comment]
I was in need of this as well.  I'll have to find a workaround.  The API i'm trying to use return a big list of ids like:

```
[
    "100D",
    "1BQK",
    "1DP5",
    "1FL5",
....
]
    ```
    
    I need to get those ids and then batch request their details on a substream but seems like that is not possible witht he low code solution.

**Comment by akozichev** [comment]
https://www.ontime360.com/ is another example, of getting a list of IDs which need to lookup individually. Will have to go with self-hosted custom solution.

[a comment by an automated account is not shown]
